### ISSUE TITLE
Books: clean 035 field

### ISSUE THREAD

**Issue by ludmilamarian** [CERNDocumentServer/cds-dojson#167]
parent of #169 
blocked by https://github.com/CERNDocumentServer/cds-migrator-kit/issues/15

Currently, these are the possible values in 035__9:

set(['CNUM-INSPIRE', 'IEEECONF', 'inspire', 'INSPIRE-CNUM', 'SCEM', 'arXiv', 'LHCLHC', 'SPIRES', 'DOE', '926408', 'ADMADM', 'Iinspire-CNUM', 'Inspire-CNUM', 'CERCER', 'INDICO.CERN.CH', 'SLACCONF', 'INIS', 'INPIRE-CNUM', 'DLC', 'HAL', 'DESY', 'FIZ', 'WAI01', 'http://inspirehep.net/oai2d', 'Isnpire-CNUM', 'INSPRIE-CNUM', 'SAFARI', 'Isnpire', 'SLAC', 'INSPIRE', 'AgendaMaker', 'KEK', 'Inspire', 'EBL', '290555', 'INSPEC', 'CERN annual report', 'CERN', '273873', 'inspire-CNUM']) but it looks like only the inspire-cnum/inspirecnum are being treated in a slightly different way.

There are several things here that I think @agentilb could help with:
i) the normalization of these values, some of them are obviously typos (`Isnpire`).
ii) it looks like inspire appears in various forms, do all relate to the same `inspire_cnum`? If yes, they should be migrated to the same filed in the new data model
iii) the cleaning of these values, as there are also number there, that might be a mistake and they need to be in another subfield
iv) only CERCER should potentially disappear (but only after doing CERNDocumentServer/cds-migrator-kit#15), or there are others that could be ignored?

**Comment by agentilb** [comment]
I have cleaned the obviously wrong occurrences (typo, wrong value, numbers...)
Remains what should stay during the migration. 
Indeed only CERCER should disappear (once the linking with old Aleph numbers is solved). 
I wonder what is ADMADM, there are several thousands of records. Not sure what it refers to and it looks like this is similar to CERCER, but one should investigate before deleting. It seems it is something similar to CERCER but for other types of records. ludmilamarian do you have a way to see if those are used in some way in the system?

Also, there is 'http://inspirehep.net/oai2d' not sure this information should be stored here. But I leave it to you to decide, it seems to me this is more a "technical information" about the harvest of the record.  

For Inspire, we can link in 2 ways: 
- Via INSPIRE-CNUM -> which is the Inspire Conference Code, in this case you find 035__9:INSPIRE-CNUM (whatever the case is)
- Via Inspire recid -> which links to the Inspire corresponding records, in this case, you find 035__9:Inspire (whatever the case is)

Hope this helps,

Anne

**Comment by ludmilamarian** [comment]
Great, thank you @agentilb !

I have recomputed the list.

```['IEEECONF', 'inspire', 'INSPIRE-CNUM', 'SCEM', 'arXiv', 'LHCLHC', 'SPIRES', 'DOE', 'ADMADM', 'Iinspire-CNUM', 'Inspire-CNUM', 'CERCER', 'INDICO.CERN.CH', 'SLACCONF', 'INIS', 'DLC', 'HAL', 'DESY', 'FIZ', 'WAI01', 'http://inspirehep.net/oai2d', 'SAFARI', 'SLAC', 'Inspire', 'AgendaMaker', 'KEK', 'INSPIRE', 'EBL', 'INSPEC', 'CERN annual report', 'inspire-CNUM']```

There was another typo for `Iinspire-CNUM` I have just fixed.

* Do you know what is the`CERN annual report` for? 
* Also, we have `SLAC`, `SPIRES`, `SLACCONF` but theoretically these have been replaced by Inspire, right?
* For `AgendaMaker` and `Indico` I will check with our colleagues if the number are still relevant.
* `LHCLHC` is just 1 record, do you think we can remove the field? https://cds.cern.ch/record/472405/export/hm?ln=en
* I will dig into `ADMADM`
* I already created a task to clean the `CERCER` (aka Aleph numbers)
* Do you have any idea what `DOE` stands for?

Thanks a lot for your help!

**Comment by ludmilamarian** [comment]
@agentilb regarding slac/spires I have just ran a quick script and there are ~ 20 records that have `SLACCONF` bot no `INSPIRE-CNUM`. All the rest of the records that mention spires/slac contain also inspire numbers. I have created a new ticket for this: https://github.com/CERNDocumentServer/cds-dojson/issues/169

**Comment by ludmilamarian** [comment]
I have recomputed the list of values (based an all records that we need to migrate), and based on the comments above these are the things that need to be done:

List of tags to keep:
`['inspire', 'ebl', 'lhclhc', 'dlc', 'hal', 'fiz', 'wai01', 'indico.cern.ch', 'scem', 'inis', 'udccer', 'ieeeconf', 'kek', 'desy', 'safari', 'inspire-cnum', 'arxiv', 'doe' ]`

List of tags to fix: 
- [ ] 'isnpire'
- [ ] '926408'
- [ ] 'agendamaker'
- [ ] 'cercer'
- [ ] 'insprie-cnum'
- [ ] '290555'
- [ ] 'isnpire-cnum'
- [ ] 'cnum-inspire'

To be decided:
'cern annual report'

To ignore:
'slac'
'slacconf'
'http://inspirehep.net/oai2d'
'spires'

**Comment by agentilb** [comment]
Strange: all the typos have been corrected on 19/10, and the correction was simply removed during the night after. 
See: 
https://cds.cern.ch/record/edit/compare_revisions?recid=113410&rev1=20181019112212&rev2=20180611195506
and
https://cds.cern.ch/record/edit/compare_revisions?recid=113410&rev1=20181019112212&rev2=20181019220255

Do you understand why?

**Comment by agentilb** [comment]
I did several multi-edits that day to clean records.. I don't remember having done one that touched so many records, but that's probably me, though. What I never realised is that multiedit modifications touch all records of the search results, even if the modification doesn't apply to them. I guess that's the cause of this.

**Comment by agentilb** [comment]
'cern annual report' can also be ignored.

**Comment by ludmilamarian** [comment]
Regarding the `agendamaker` and records that potentially are lectures, I will summarize here my findings:

Our colleagues from Indico gave us the corresponding new IDs, however, for past videos I think we can not replace them yet, because some recordings might be stored in folders named based on the old ID. This will be sorted out when we migrate lectures.

I think we need to see what we should migrate as part of books, and then do changes only on these records. 

- [ ] Currently there are 23 records that are streaming videos in the proceedings collection:
https://cds.cern.ch/search?ln=en&cc=Books+%26+Proceedings&p=340__a%3A%22Streaming+video%22
I think they should be filtered, as except the last one that looks indeed like a proceedings, the rest look more like lectures.

- [ ] There are 26 records pointing to old events in indico (agendamaker):  https://cds.cern.ch/search?ln=en&cc=Books+%26+Proceedings&sc=1&p=035__9%3A%22AgendaMaker%22

Once you decide which of these records should be kept, I can replace the previous `agendamaker` IDs with `Indico` IDs.  Also, I see now that some records are mixing video recordings and proceedings. I think if we plan to migrate some of these records we should create separate records for the proceeding and for the video. The records can link with each other, but one record can not be both a video and a proceeding. Let me know what you think.

**Comment by agentilb** [comment]
For: 26 records pointing to old events in indico (agendamaker): https://cds.cern.ch/search?ln=en&cc=Books+%26+Proceedings&sc=1&p=035__9%3A%22AgendaMaker%22

Can you let me know the new Indico ids of those events, so I can check what content is on Indico, and decide what are real proceedings or not? Indeed, I guess most of them will need to be filtered out...

**Comment by ludmilamarian** [comment]
http://cds.cern.ch/record/517300 -> http://indico.cern.ch/event/408840
http://cds.cern.ch/record/726659 -> http://indico.cern.ch/event/041075
http://cds.cern.ch/record/733606 -> http://indico.cern.ch/event/041428
http://cds.cern.ch/record/741349 -> http://indico.cern.ch/event/415024
http://cds.cern.ch/record/801691 -> http://indico.cern.ch/event/419824
http://cds.cern.ch/record/821452 -> http://indico.cern.ch/event/042407
http://cds.cern.ch/record/822407 -> http://indico.cern.ch/event/419871
http://cds.cern.ch/record/931361 -> http://indico.cern.ch/event/411250
http://cds.cern.ch/record/933801 -> http://indico.cern.ch/event/427826
http://cds.cern.ch/record/933801 -> http://indico.cern.ch/event/427833
http://cds.cern.ch/record/933801 -> http://indico.cern.ch/event/427867
http://cds.cern.ch/record/933801 -> http://indico.cern.ch/event/427929
http://cds.cern.ch/record/933801 -> http://indico.cern.ch/event/427948
http://cds.cern.ch/record/933801 -> http://indico.cern.ch/event/427973
http://cds.cern.ch/record/933801 -> http://indico.cern.ch/event/427975
http://cds.cern.ch/record/935121 -> http://indico.cern.ch/event/432271
http://cds.cern.ch/record/935131 -> http://indico.cern.ch/event/411417
http://cds.cern.ch/record/973361 -> http://indico.cern.ch/event/417197
http://cds.cern.ch/record/973362 -> http://indico.cern.ch/event/417832
http://cds.cern.ch/record/973364 -> http://indico.cern.ch/event/417559
http://cds.cern.ch/record/973372 -> http://indico.cern.ch/event/417574
http://cds.cern.ch/record/974801 -> http://indico.cern.ch/event/419450
http://cds.cern.ch/record/974803 -> http://indico.cern.ch/event/419817
http://cds.cern.ch/record/974806 -> http://indico.cern.ch/event/420145
http://cds.cern.ch/record/975187 -> http://indico.cern.ch/event/423040
http://cds.cern.ch/record/975421 -> http://indico.cern.ch/event/424522
http://cds.cern.ch/record/975439 -> http://indico.cern.ch/event/424234
http://cds.cern.ch/record/975440 -> http://indico.cern.ch/event/424677
http://cds.cern.ch/record/976076 -> http://indico.cern.ch/event/427250
http://cds.cern.ch/record/976076 -> http://indico.cern.ch/event/427251
http://cds.cern.ch/record/977394 -> http://indico.cern.ch/event/427321
http://cds.cern.ch/record/987830 -> http://indico.cern.ch/event/409154

**Comment by agentilb** [comment]
From the list above, we keep for sure:
http://cds.cern.ch/record/517300 -> http://indico.cern.ch/event/408840
http://cds.cern.ch/record/741349 -> http://indico.cern.ch/event/415024
http://cds.cern.ch/record/821452 -> http://indico.cern.ch/event/042407
http://cds.cern.ch/record/822407 -> http://indico.cern.ch/event/419871

For:
http://cds.cern.ch/record/726659 -> http://indico.cern.ch/event/041075
http://cds.cern.ch/record/733606 -> http://indico.cern.ch/event/041428
-> they are not anymore in the Proceedings collection.

The others are events where there is only transparencies and/or videos, and in many cases, the Indico page has a restricted access. They shouldn't be in the proceedings collection, I believe.

**Comment by ludmilamarian** [comment]
@agentilb what do you think we do with these records? should we update the 4 that should remain in the proceedings and move the rest in the lectures collections, or do do you have another suggestion?

**Comment by agentilb** [comment]
I've indeed removed the AgendaMaker id  and added the link from the video to the proceedings for the 2 records:  https://cds.cern.ch/record/821452 and https://cds.cern.ch/record/933801

For: https://cds.cern.ch/record/741349?ln=en if I'm not mistaken, there is no video on CDS and the link seems broken from Indico, do you think something is retrievable from AgendaMaker? Should we keep it.
For: https://cds.cern.ch/record/517300?ln=en
and https://cds.cern.ch/record/822407 there is no video at all on Indico and we do not have the DVD, do you think the videos are somewhere? If no, I would delete AgendaMaker and the Video information. 

For the DVD holdings, I don't think we decided to delete them, we would need to do some cleaning beforehand, and this will be time consuming. Is it a problem to migrate them?

**Comment by ludmilamarian** [comment]
Hi @agentilb From my side, I think it is ok to close this ticket. I have update 2 or 3 records from agendaMaker to new Indico IDs. I think the DVDs issue is handled in another ticket. Let me know if there is still something to be addressed here.

**Comment by agentilb** [comment]
I noticed one more thing for the 035 field. 
for the arxiv OAI information stored in 035 such as in: 
https://cds.cern.ch/record/edit/?ln=en#state=edit&recid=2240869
do you confirm this can be ignored during the migration or should it be stored somewhere? 

Otherwise, I thing we can close the ticket.

**Comment by ludmilamarian** [comment]
@agentilb do you want to ignore completely any 035 that has `035$$9arXiv` ? Or we should keep $$9 and $$a and ignore the rest. We currently keep $$9 and $$a, but there is nothing defined for the rest as far as I can see.

**Comment by agentilb** [comment]
If you do not need on your side, I think we can ignore completely 035 when $$9.arxiv  since the arxiv number is normally stored  in 037.

**Comment by ludmilamarian** [comment]
Makes sense @agentilb ! I have created a new ticket for this https://github.com/CERNDocumentServer/cds-dojson/issues/196 just because this one is getting bigger and bigger. If this was the only thing left, then I will close this one.
